src/client/auth.js

```diff
@@ -1,16 +1,67 @@
 const UUID = require('uuid-1345')
 const yggdrasil = require('yggdrasil')
+const fs = require('mz/fs')
+const promisify = require('es6-promisify')
+const mkdirp = promisify(require('mkdirp'))
 
-module.exports = function (client, options) {
+module.exports = async function (client, options) {
   const yggdrasilClient = yggdrasil({ agent: options.agent, host: options.authServer || 'https://authserver.mojang.com' })
-  const clientToken = options.clientToken || (options.session && options.session.clientToken) || UUID.v4().toString()
+  const clientToken = (options.profilesFolder && (await getLP()).clientToken) || options.clientToken || (options.session && options.session.clientToken) || UUID.v4().toString().replace("-", "")
   const skipValidation = false || options.skipValidation
   options.accessToken = null
-  options.haveCredentials = options.password != null || (clientToken != null && options.session != null)
+  options.haveCredentials = options.password != null || (clientToken != null && options.session != null) || options.profilesFolder != null
+
+  async function getLP () { // get launcher profile
+    let data
+    try {
+      data = JSON.parse(await fs.readFile(options.profilesFolder + '/launcher_profiles.json', 'utf8'))
+    } catch (err) {
+      await mkdirp(options.profilesFolder)
+      await fs.writeFile(options.profilesFolder + '/launcher_profiles.json', '')
```

Comment:
Is it a good idea to create an empty json file ?

Reply:
I think so, yeah. It was something suggested in the issue that if it doesn't exist, it creates it. Otherwise this behavior can be easily changed of course

Reply:
I mean more, maybe create it with an empty object inside like `{}` because an empty file is not a valid json file